python/lsst/daf/butler/cli/cmd/_remove_runs.py

```diff
@@ -0,0 +1,104 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import click
+from typing import Mapping, Sequence
+
+from ..utils import (
+    ButlerCommand,
+)
+from ..opt import (
+    collection_argument,
+    confirm_option,
+    options_file_option,
+    repo_argument,
+)
+from ... import script
+
+
+# messages emitted by remove-runs, defined separately for use in unit
+# tests.
+noRunCollectionsMsg = "No RUN collections were found."
+willRemoveRunsMsg = "The following RUN collections will be removed:"
+willRemoveDatasetsMsg = "The following datasets will be removed:"
+didRemoveRunsMsg = "The following RUN collections were removed:"
+didRemoveDatasetsMsg = "The following datasets were removed:"
+removedRunsMsg = "Removed collections"
+abortedMsg = "Aborted."
+noRepoFoundMsg = "No repo was found at '{root}'"
+
+
+def _print_remove(will: bool, runs: Sequence[str], datasets: Mapping[str, int]):
+    """Print the formatted remove statement.
+
+    Parameters
+    ----------
+    will : bool
+        True if remove "will" happen, False if the remove "did" happen.
+    runs : Sequence[str]
+        The RUNs that will be or were removed.
+    datasets : Mapping[str, int]
+        The dataset types & count that will be or were removed.
+    """
+    print(f"\n{willRemoveRunsMsg if will else didRemoveRunsMsg}\n")
+    print(", ".join(runs))
+    print(f"\n{willRemoveDatasetsMsg if will else didRemoveDatasetsMsg}\n")
+    print(", ".join([f"{i[0]}({i[1]})" for i in datasets.items()]))
+    print()
+
+
+@click.command(cls=ButlerCommand)
+@repo_argument(required=True)
+@collection_argument(
+    help="COLLECTION is a glob-style expression that identifies the RUN collection(s) to remove."
+)
+@confirm_option()
+@options_file_option()
+def remove_runs(**kwargs):
+    """Remove one or more RUN collections.
+
+    This command can be used to remove RUN collections and the datasets within
+    them.
+    """
+    confirm = kwargs.pop("confirm")
+    try:
+        result = script.removeRuns(**kwargs)
+    except FileNotFoundError:
```

Comment:
This seems inconsistent with every other command where we let the FileNotFoundError continue to the caller. Why is this different?

Reply:
Just trying to be more complete. Maybe it's better to be consistent with other existing commands? Qserv folks were complaining when the CLI tools allowed errors to continue to the caller. I find it useful sometimes to have _all_ the info, but others seem to think it's messy, not useful. I'll remove the handling to be consistent, unless you think it should stay.

Reply:
If it stays you still have to fix the exit status. I'd be inclined to remove it for now. Andy already added some special code to limit the amount of stack trace that is reported from these commands.